### PR TITLE
276 test reproducibility of build with open data

### PR DESCRIPTION
**Pull request by wangzhao0217**
- add simplify_network to tar_target
- add simplify_network to _targets.R with reproducibility of data

### CODE REVIEW

File: _targets.R [hunk 1]
```diff
@@ -1197,7 +1197,15 @@ tar_target(pmtiles_rnet, {
   }),
 
   tar_target(simplify_network, {
-
+    # remotes::install_dev("stplanr")
+    packageVersion("stplanr")
+    sf::sf_use_s2(TRUE)
+    library(stplanr)
+    library(dplyr)
```
Comment (Robinlovelace) [inline _targets.R:1204]: You can remove these tests when it's ready to merge.

**Review by Robinlovelace — COMMENTED** [review]
Looks like good progress, did you get it working?

**Comment by wangzhao0217** [comment]
Hi @Robinlovelace ,

The targets not using the latest stplanr, but I am not sure how to update it. 
I have tested it on two pc, and successful run requires importing the following library:  
    remotes::install_dev("stplanr")
    library(stplanr)
    library(dplyr)

**Comment by Robinlovelace** [comment]
> Hi @Robinlovelace ,
> 
> The targets not using the latest stplanr, but I am not sure how to update it. I have tested it on two pc, and successful run requires importing the following library: remotes::install_dev("stplanr") library(stplanr) library(dplyr)

@wangzhao0217 fine to have those lines of code. I'm just suggesting that they go in a different place, e.g. at the top of `_targets.R` around here:

https://github.com/nptscot/npt/blob/1931fb2bff7f9c20e10f9aedf93f37fbaebc5d09/_targets.R#L17-L21

Does that make sense and sound reasonable?

File: _targets.R [hunk 2]
```diff
@@ -19,7 +19,9 @@ library(magrittr) # Light load of %>%
 library(sf)
 library(future) # Needed for multi-core running
 library(future.callr)
-
+remotes::install_dev("stplanr", force = TRUE)
```
Comment (Robinlovelace) [inline _targets.R:22]: :+1:

**Review by Robinlovelace — COMMENTED** [review]
Looking good.

**Comment by Robinlovelace** [comment]
@wangzhao0217 I'd like to get this merged early next week. Do you think it's ready? Hope you had a good week off and catch up next week.

**Review by Robinlovelace — COMMENTED** [review]
How did it go?

**Comment by Robinlovelace** [comment]
Heads-up @wangzhao0217 I'm seeing this message when trying to debug:

```
✖ Assigned data has 0 rows.
ℹ Only vectors of size 1 are recycled.
Caused by error in `vectbl_recycle_rhs_rows()`:
! Can't recycle input of size 0 to size 474.
Run `rlang::last_trace()` to see where the error occurred.
Warning messages:
1: attribute variables are assumed to be spatially constant throughout all geometries 
2: In st_cast.sf(sf::st_cast(x, "MULTILINESTRING"), "LINESTRING") :
  repeating attributes for all sub-geometries for which they may not be constant
3: st_centroid assumes attributes are constant over geometries 
```

**Comment by Robinlovelace** [comment]
At this line after loading the necessary objects: 

https://github.com/nptscot/npt/blob/ddfb98873940fb8f66cc040bdad6031a6cf00a05/_targets.R#L1241

Not sure why.

**Comment by Robinlovelace** [comment]
Tried with projected data, no luck.

**Comment by Robinlovelace** [comment]
Working yet @wangzhao0217 ?

**Review by Robinlovelace — COMMENTED** [review]
How did the latest changes go? Fingers crossed!

**Comment by wangzhao0217** [comment]
I got the admin right to install, it solved several error,but the rest is
"university of Leeds caused windows security blocks the installation
action". I will ask IT on Monday to disable the windows security feature
for a day to let me finish the installation.

**Comment by Robinlovelace** [comment]
OK. How about the new target on the 'open data build' on your laptop?

**Comment by wangzhao0217** [comment]
Hello Robin, I have tested it and it works. Please take a look at the
latest push. I have asked IT to temporarily disable the Windows security
feature to install odjitter as well.

**Comment by Robinlovelace** [comment]
Good work. Will test when I get a chance..

**Comment by Robinlovelace** [comment]
There are 2 new targets here. Will merge.

File: _targets.R [hunk 3]
```diff
@@ -11,22 +11,21 @@ if(FALSE){ # Repeated builds can it GitHub API limit, set to TRUE to check for p
...
 library(future) # Needed for multi-core running
 library(future.callr)
-
+library(stplanr)
+library(dplyr)
 # Set target options:
 pkgs = packages = c(
-  "tibble","zonebuilder","dplyr","stplanr","lubridate",
-  "cyclestreets","stringr","sf","tidyr","data.table",
+  "tibble","zonebuilder","dplyr","lubridate",
```
Comment (Robinlovelace) [inline _targets.R:25]: Important change removes CRAN version of stplanr. Heads-up @mem48 and @joeytalbot this fixes the build.

**Review by Robinlovelace — APPROVED** [review]

**Pull request merged**